integreat_cms/deepl_api/utils.py

```diff
@@ -0,0 +1,161 @@
+        self, request, content_translations, language_slug, form_class, content_label
+    ):
+        """
+        This functions gets the translation from DeepL
+
+        :param request: passed request
+        :type request: ~django.http.HttpRequest
+
+        :param content_translations: passed content
+        :type content_translations: ~django.db.models.query.QuerySet [ ~integreat_cms.cms.models.abstract_content_translation.AbstractContentTranslation ]
+
+        :param language_slug: current GUI language slug
+        :type language_slug: str
+
+        :param form_class: passed Form class of content type
+        :type form_class: ~integreat_cms.cms.forms.custom_content_model_form.CustomContentModelForm
+
+        :param content_label: passed content type, either event or poi
+        :type content_label: str
+        """
+        # Get target language
+        target_language = request.region.get_language_or_404(language_slug)
+        source_language = request.region.get_source_language(language_slug)
+        for content_translation in content_translations:
+            source_translation = content_translation.get_translation(
+                source_language.slug
+            )
+            existing_target_translation = content_translation.get_translation(
+                target_language.slug
+            )
+            # For some languages, the DeepL client expects the BCP tag instead of the short language code
+            if target_language.slug in ("en", "pt"):
+                target_language_key = target_language.bcp47_tag
+            else:
+                target_language_key = target_language.slug
+            target_title = self.get_translation(
+                source_translation.title,
+                source_language.slug,
+                target_language_key,
+            )
+            data = {
+                "title": target_title,
+                "status": existing_target_translation.status
+                if existing_target_translation
+                else source_translation.status,
+            }
+            if source_translation.content:
+                data["content"] = self.get_translation(
+                    source_translation.content,
+                    source_language.slug,
+                    target_language_key,
+                )
+            # for pois adds a short description
+            if content_label == "poi":
+                data["short_description"] = self.get_translation(
+                    source_translation.short_description,
+                    source_language.slug,
+                    target_language_key,
+                )
+            content_translation_form = form_class(
+                data=data,
+                instance=existing_target_translation,
+                additional_instance_attributes={
+                    "creator": request.user,
+                    "language": target_language,
+                    content_label: content_translation,
```

Comment:
```suggestion
                    source_translation.foreign_field(): content_object,
```
The content label is already stored in the translation model, so no need to pass it to this function.